lib/graphql_devise/schema_plugin.rb

```diff
@@ -79,7 +79,11 @@ def authenticate_option(field, trace_data)
       auth_required = if trace_data[:context]
         field.metadata[:authenticate]
       else
-        if Gem::Version.new(GraphQL::VERSION) >= Gem::Version.new('1.13.1')
+        if Gem::Version.new(GraphQL::VERSION) >= Gem::Version.new('2.0')
+          # authenticate will not be defined if field_class is not defined in the type
```

Comment:
```suggestion
          # authenticate will only be defined if "field_class GraphqlDevise::Types::BaseField" is added to the type
```

Reply:
Yes, this definitively needs a refractor. I think that for v1 we could remove support for gql 1.8 which I think is the outer if. Mmm and I also could remove the silence depreciation condition since we already provide an alternative. Might need to be explicit about the depreciation message though, since the gems depreciation message will not be clear about how to fix it